diff -up system-config-printer-1.1.16/jobviewer.py.escape-markup system-config-printer-1.1.16/jobviewer.py
--- system-config-printer-1.1.16/jobviewer.py.escape-markup	2010-05-19 16:59:10.000000000 +0200
+++ system-config-printer-1.1.16/jobviewer.py	2010-05-19 17:04:42.000000000 +0200
@@ -38,6 +38,7 @@ import subprocess
 import sys
 import time
 import urllib
+from xml.sax import saxutils
 
 from debug import *
 import config
@@ -1346,7 +1347,7 @@ class JobViewer (GtkGUI, monitor.Watcher
 
                 markup = ('<span weight="bold" size="larger">' +
                           _("Print Error") + '</span>\n\n' +
-                          message)
+                          saxutils.escape (message))
                 try:
                     if event['printer-state'] == cups.IPP_PRINTER_STOPPED:
                         name = event['printer-name']
