diff -up system-config-printer-1.1.16/statereason.py.statereason-localized system-config-printer-1.1.16/statereason.py
--- system-config-printer-1.1.16/statereason.py.statereason-localized	2010-06-07 18:26:27.000000000 +0200
+++ system-config-printer-1.1.16/statereason.py	2010-06-07 18:30:43.000000000 +0200
@@ -143,10 +143,10 @@ class StateReason:
                     schemes = ["text", "http", "help", "file"]
                     localized_reason = ""
                     for scheme in schemes:
-                        reason = self._ppd.localizeIPPReason(self.reason,
-                                                             scheme)
-                        if reason != None:
-                            localized_reason = localized_reason + reason + ", "
+                        lreason = self._ppd.localizeIPPReason(self.reason,
+                                                              scheme)
+                        if lreason != None:
+                            localized_reason = localized_reason + lreason + ", "
                     if localized_reason != "":
                         reason = localized_reason[:-2]
                 except RuntimeError:
