diff -up system-config-printer-1.1.16/system-config-printer.py.npinit-traceback system-config-printer-1.1.16/system-config-printer.py
--- system-config-printer-1.1.16/system-config-printer.py.npinit-traceback	2010-01-06 15:38:44.618286232 +0000
+++ system-config-printer-1.1.16/system-config-printer.py	2010-01-06 15:38:54.800408801 +0000
@@ -3974,7 +3974,7 @@ class NewPrinterGUI(GtkGUI):
                     self.auto_make = devid_dict["MFG"]
                     self.auto_model = devid_dict["MDL"]
                     self.auto_driver = None
-                if not self.device or not self.device.id:
+                if self.device and not self.device.id:
                     self.device.id = devid
                     self.device.id_dict = cupshelpers.parseDeviceID (devid)
                 self.mainapp.devid = ""
