diff -up system-config-printer-1.1.16/jobviewer.py.state-reason-blacklist system-config-printer-1.1.16/jobviewer.py
--- system-config-printer-1.1.16/jobviewer.py.state-reason-blacklist	2010-06-22 16:45:15.038999004 +0100
+++ system-config-printer-1.1.16/jobviewer.py	2010-07-19 15:38:55.579514086 +0100
@@ -1151,6 +1151,20 @@ class JobViewer (GtkGUI, monitor.Watcher
             # Not important enough to justify a notification.
             return
 
+        blacklist = [
+            # Some printers report 'other-warning' for no apparent
+            # reason, e.g.  Canon iR 3170C, Epson AL-CX11NF.
+            # See bug #520815.
+            "other",
+
+            # This seems to be some sort of 'magic' state reason that
+            # is for internal use only.
+            "com.apple.print.recoverable",
+            ]
+
+        if reason.get_reason () in blacklist:
+            return
+
         self.notify_printer_state_reason (reason)
 
     def notify_printer_state_reason (self, reason):
@@ -1159,9 +1173,6 @@ class JobViewer (GtkGUI, monitor.Watcher
             debugprint ("Already sent notification for %s" % repr (reason))
             return
 
-        if reason.get_reason () == "com.apple.print.recoverable":
-            return
-
         level = reason.get_level ()
         if (level == StateReason.ERROR or
             reason.get_reason () == "connecting-to-device"):
